### ISSUE TITLE
Incorrect subscription expired message

### ISSUE THREAD

**Issue by HarelM** [IsraelHikingMap/Site#1763]
## What I expect to happen
I expect to get an expired subscription error message only when a subscription expired

## What really happened
I get the message when the subscription was not expired

## What I did that caused the issue - step by step
1. Have a valid subscription
2. Try download offline files when there's no network for example

## My environment
* Operating System and version: Mobile, any
* Web Browser and version: N/A

## Things I think the developers should know, images, links etc.
This is related to #1699 and this bug was introduced as part of the fix there, unfortunately.

**Comment by HarelM** [comment]
Phone call conversation summary:
1. Try and validate the subscription using our server in parallel to the purchase plugin code
2. If the server returns a valid result or an error treat is accordingly
3. If the server doesn't return a value and the store doesn't return a value use the last state

This is hard to test though, I'm not sure what would be an easy way to test this. One option I thought of is maybe try logging in with a different OSM account (that doesn't have a valid subscription) on a phone that has a valid subscription, not sure what will happen, but it might trigger the right code in the server side, maybe, IDK...

**Comment by HarelM** [comment]
Further info related to this issue:
Server behaviour:
1. I've tested the fovea API against a user that has an expired subscription and the response code is 200, the payload has expired true in it. In that case our server will return 403 - forbidden when requesting the files to download
2. If fovea API returns another status code other than 200 or if the request to the fovea API times out, our server will respond with 500 - internal server error.
3. If everything is OK our server will return 200 and the relevant files that needs to be downloaded.

I've moved the logic of if there's a need to renew subscription to when the app starts and in all cases of an issue when trying to download files the client will ask the user to try again later (i.e oops, something went wrong...).

When the app starts it will check what is the status against our server:
1. In case of timeout, client side error or server side non-403 error: it will keep the current state (which ever it was before)
2. In case of success it will set the state to offline available (i.e. the button will show download and not purchase)
3. In case of server error 403 it will change the status to offline unavailable (i.e. purchase now) and the next click will open the store to allow purchasing the subscription

I've also added more logging in the server and in the client to better understand future problems in this flow.
I hope this concludes this issue.
@zstadler let me know if I missed anything.
